添加工具箱页并新增切换新旧配置功能

YvonneOfficial opened the pull request:
在程序中添加了工具箱页，并且在工具箱页内添加了一个新的功能
切换新旧配置 用于适配新旧版本的Among Us来保证不会出现黑屏问题

## Sourcery 总结

在应用程序中添加了一个新的工具页面，并为《Among Us》实现了一个配置切换器，用于在旧版和当前设置之间进行切换，并已集成到主导航中。

新功能：
- 引入了一个 `ToolsInterface` 视图，其中包含一个 `SettingCard`，用于在《Among Us》的旧配置和新配置之间进行切换
- 提供备份、加载、应用和保存 `settings.amogus` 的功能，适用于旧版和更新后的游戏版本

改进：
- 将工具页面集成到主窗口导航中，并从主界面进行路由

<details>
<summary>Original summary in English</summary>

```markdown
## Sourcery 总结

添加一个新的工具页面，用于切换 Among Us 配置以支持跨版本兼容性，并将其集成到主导航中，同时实现底层的备份、加载、应用和保存文件操作。

新功能：
- 引入带有 SwitchConfigCard 的 ToolsInterface 视图，用于在旧版和更新的配置之间切换 Among Us 设置
- 实现对 settings.amogus 文件的备份、加载、应用（新/旧）和保存操作

改进：
- 将工具箱页面集成到主窗口导航中，并支持从主界面进行导航
- 通过移除手动事件刷新并优化日志消息，清理启动画面的启动逻辑
```

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add a new tools page to switch Among Us configuration to support compatibility across versions, integrating it into the main navigation and implementing the underlying file operations for backup, load, apply, and save.

New Features:
- Introduce ToolsInterface view with SwitchConfigCard to toggle Among Us settings between legacy and updated configurations
- Implement backup, load, apply (new/old), and save operations for settings.amogus file

Enhancements:
- Integrate toolbox page into main window navigation and enable navigation from the home interface
- Cleanup splash screen startup logic by removing manual event flushing and refine log messages

</details>

</details>

YvonneOfficial committed:
添加工具箱页并新增切换新旧配置功能

Hiraeth-Wave commented on app/view/toolsInterface.py:
```diff
@@ -0,0 +1,203 @@
+    "normalSearchOptions": "CoQAAAEAAAoAAAEAdwAAgD8AAIA/AADAPwAAcEEBAQIBAAAAAwEPAAAAeAAAAAEAAQEAAAAJBQAAAAMAAAAKCAIAAAACAAAPBQQAAAADAAA8CgADAAAAAgAAHg8IAAAAAgAACgEJAAAAAgAADx4KAAAAAwAADx4BDAAAAAEAAAMSAAAAAQAADw==",
+    "hideNSeekHostOptions": "CkIAAAIAAA8AAQAAAAAAgD8AAIA/AADAPwEBAgEBAAAAAABIQzMzsz4AAIA+AQEAAEhCmpmZPwEA/////wAAwEAAAEBAAA==",
+    "hideNSeekSearchOptions": "CkIAAAIAAA8AAQAAAAAAgD8AAIA/AADAPwEBAgEBAAAAAABIQzMzsz4AAIA+AQEAAEhCmpmZPwEA/////wAAwEAAAEBAAA=="
+}
```
依据[开发规范](https://github.com/QingFengTechnology/FengAmongUsTool/tree/v4?tab=contributing-ov-file#%E5%BC%80%E5%8F%91%E8%A7%84%E8%8C%83)，配置内容应保存在[variableConfig.py](https://github.com/QingFengTechnology/FengAmongUsTool/blob/v4/app/function/variableConfig.py)。

Hiraeth-Wave commented on app/module/mainWindow.py:
```diff
@@ -161,6 +152,10 @@ def initNavigation(self):
 
+        # 添加工具箱界面到导航滚动区
+        self.addSubInterface(
+            self.toolsInterface, FIF.APPLICATION, '工具箱', NavigationItemPosition.SCROLL)
```
将`工具箱`重命名为`工具集合`以与主页卡片名称统一，同时使导航栏排版更美观。

Hiraeth-Wave commented on app/module/mainWindow.py:
```diff
@@ -1,7 +1,9 @@
+#!/usr/bin/env python3
```
不需要的标记，请撤销。
——我们的贡献指南推荐开发者使用虚拟环境开发此项目。

Hiraeth-Wave commented on app/view/toolsInterface.py:
```diff
@@ -0,0 +1,203 @@
+            self._showInfoBar('error', '错误', '无法找到原始配置文件进行备份')
+        except Exception as e:
+            logger.exception('备份 settings.amogus 失败')
+            self._showInfoBar('error', '错误', f'备份配置失败: {e}')
```
以此行代码为例，通常来说不建议在一个小小的提示框里显示原始错误信息，我们有日志系统且会输出到文件。
此评论仅作建议，如果没有被更改我会新打开一个 PR 解决。

YvonneOfficial committed:
根据代码审查评论中的建议修改代码

Hiraeth-Wave commented on app/module/configSwitcher.py:
根据[开发规范](https://github.com/QingFengTechnology/FengAmongUsTool/tree/v4?tab=contributing-ov-file#%E5%BC%80%E5%8F%91%E8%A7%84%E8%8C%83)：
- 函数应使用[小驼峰命名法](https://baike.baidu.com/item/lowerCamelCase/18434330)。
- 对于只有单个功能使用的函数，应放置在`/app/module`文件夹。

——既然使用了 AI，就请正确使用它。

Hiraeth-Wave commented on app/module/mainWindow.py:
为什么你要删除这些注释？

QingFeng-awa reviewed (changes requested):
在功能测试中发现切换配置后会创建一个备份文件，且看起来程序不会删除该备份文件(或是缺少了相关的日志输出)，且**随着使用次数增多这些文件也会越积越多**，显然这不是我们期望的。
请对其进行优化，例如将其在不再使用时删除，如果备份文件会被用作发生意外时恢复以外的用途，也请让其的数量保持在一个文件。

YvonneOfficial committed:
再次根据代码审查评论中的建议修改代码。
- 恢复了本不应该删除的注释，这是我的疏忽，十分抱歉。
- 现在程序在切换配置后会自动删除所有旧的备份并创建新备份。
- 现在的configSwitcher.py存放于app\module，并在命名函数时使用了小驼峰命名法。

Hiraeth-Wave commented on app/module/configSwitcher.py:
```diff
@@ -41,16 +41,26 @@ def save_settings(path: Path, data: dict) -> None:
     shutil.copy2(source, backup_path)
     logger.info('配置备份已创建: %s', backup_path)
     return backup_path
 
```
不一定非要使用随机生成的文件名用作备份文件，万一有哪个工具箱和我们的逻辑撞了呢(
可以考虑改为固定后缀，你总会清理的文件固定文件名还方便点。

> 仅作建议，但这个对话卡住了 PR 合并进程，如果不想解决或许可以直接转为已解决对话。
